de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetRoleManagement.java

```diff
@@ -130,10 +147,17 @@ protected void createTableColumns(EditingDomain editingDomain) {
 			@Override
 			public void mouseDown(MouseEvent e) {
 				if (tableViewer.getTable().getColumnCount() > 1 && e.x > tableViewer.getTable().getColumn(0).getWidth()) {
-					openCustomDialog();
+					// Check if the user has the required permission before opening the custom dialog
+		            if (hasPermission()) {
+		                openCustomDialog();
+		            } else {
+		                // Display a message or take appropriate action for users without permission
+		                System.out.println("You don't have permission to perform this action.");

```

Comment:
We should have println statements in the file code... Either it should be send to the log or shown via UI... I guess in this case we don't need to notify the user, as there is no button but just the table